GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.cs

```diff
@@ -133,118 +144,125 @@ protected override void PageToSettings()
             }
         }
 
-        private void GlobalMergeTool_TextChanged(object sender, EventArgs e)
+        private string BrowseDiffMergeTool(string toolName, string path)
         {
-            if (IsLoadingSettings)
+            DiffMergeToolConfiguration diffMergeToolConfig = default;
+            if (_NO_TRANSLATE_GlobalDiffTool.SelectedIndex > -1)
             {
-                return;
+                diffMergeToolConfig = _diffMergeToolConfigurationManager.LoadDiffMergeToolConfig(toolName, null);
             }
 
-            var mergeTool = _NO_TRANSLATE_GlobalMergeTool.Text.Trim();
-            MergetoolPath.Text = CurrentSettings.GetValue($"mergetool.{mergeTool}.path");
-            MergeToolCmd.Text = CurrentSettings.GetValue($"mergetool.{mergeTool}.cmd");
-
-            MergeToolCmd.Enabled = !string.IsNullOrEmpty(MergeToolCmd.Text);
+            string initialDirectory = GetInitialDirectory(path.EnsureTrailingPathSeparator(),
```

Comment:
Both path and diffMergeToolConfig.Path may be the full path to the executable, this method does not check that the path is a directory

Reply:
I'm due to write tests for this functionality

Reply:
Done. The code is updated to cater for both directory and file